packages/engine-render/src/components/sheets/extensions/font.ts

```diff
@@ -103,11 +104,36 @@ export class Font extends SheetExtension {
         // fontMatrix.forValue((row: number, col: number, fontsConfig: IFontCacheItem) => {
         //     this.renderFontByCellMatrix(renderFontContext, row, col, fontsConfig);
         // });
+
+        const mergeRanges: IRange[] = [];
         viewRanges.forEach((range) => {
             range.startColumn -= EXPAND_SIZE_FOR_RENDER_OVERFLOW;
             range.endColumn += EXPAND_SIZE_FOR_RENDER_OVERFLOW;
             range = clampRange(range);
             Range.foreach(range, (row, col) => {
+                const cellInfo = spreadsheetSkeleton.getCellByIndexWithNoHeader(row, col);
+                if (!cellInfo) return;
+
+                // put all merged cell to another pass to handle.
+                if (cellInfo.isMerged || cellInfo.isMergedMainCell) {
+                    // to check if mergeRanges has this merge range already. if not, push it.
+                    const f = mergeRanges.filter((r: IRange) => {
```

Comment:
should we make sure this part time complex is O(1), O(N) maybe cause  performance issue